### PR TITLE
fix: correct types for node16 resolution

### LINKED ISSUE #46: Bundled types fail under "node16" module
# Repro

```bash
$ npm i eslint-visitor-keys typescript
$ echo '{"compilerOptions":{"strict":true,"module":"Node16","moduleResolution":"node16"}}' > tsconfig.json
$ echo "import * as foo from 'eslint-visitor-keys';" > test.ts
$ npx tsc -b ./tsconfig.json
```

# Expected

No errors

# Actual

```
test.ts:1:22 - error TS7016: Could not find a declaration file for module 'eslint-visitor-keys'. '/Users/bjz/temp/playground/node_modules/eslint-visitor-keys/dist/eslint-visitor-keys.cjs' implicitly has an 'any' type.
  Try `npm i --save-dev @types/eslint-visitor-keys` if it exists or add a new declaration (.d.ts) file containing `declare module 'eslint-visitor-keys';`

1 import * as foo from 'eslint-visitor-keys';
                       ~~~~~~~~~~~~~~~~~~~~~


Found 1 error.
```

# Additional Info

https://arethetypeswrong.github.io/?p=eslint-visitor-keys%403.4.0
image

### PR DESCRIPTION
**Pull request by bradzacher**
Fixes #46

1) Adds a rollup step to generate a `.d.cts` file for the `.cjs` bundle
2) Fixes the unnecessary type cycle in `lib` files

# Test plan

`npm pack` and upload the tarball to arethetypeswrong:
image

Regarding the "masquerading" issue reported - it shouldn't cause a problem technically: https://github.com/arethetypeswrong/arethetypeswrong.github.io/issues/21
Ideally the output files should be `.mjs`/`.d.mts` to be explicit, but it is okay without it (it's more of a pedantic report from my understanding).

I tested this works as expected by continuing the exmaple in #46:
```bash
$ echo "import * as foo from 'eslint-visitor-keys';\nexport default foo.shoulderror;" > test2.mts
$ echo "import * as foo from 'eslint-visitor-keys';\nexport default foo.shoulderror;" > test3.cts
$ echo "import foo = require('eslint-visitor-keys');\nexport default foo.shoulderror;" > test4.cts
$ npx tsc -b ./tsconfig.json
```
And it worked as expected and correctly resolved the types.

### CODE REVIEW

File: eslint-visitor-keys-3.4.0.tgz
Comment (mdjermanovic) [inline eslint-visitor-keys-3.4.0.tgz:1]: This file is probably accidentally checked in.

**Comment by mdjermanovic** [comment]
<details>

<summary>eslint-visitor-keys.d.cts</summary>

```ts
type VisitorKeys$1 = {
    readonly [type: string]: readonly string[];
};
/**
 * @typedef {{ readonly [type: string]: ReadonlyArray<string> }} VisitorKeys
 */
/**
 * @type {VisitorKeys}
 */
declare const KEYS: VisitorKeys$1;

/**
 * Get visitor keys of a given node.
 * @param {object} node The AST node to get keys.
 * @returns {readonly string[]} Visitor keys of the node.
 */
declare function getKeys(node: object): readonly string[];
/**
 * Make the union set with `KEYS` and given keys.
 * @param {VisitorKeys} additionalKeys The additional keys.
 * @returns {VisitorKeys} The union set.
 */
declare function unionWith(additionalKeys: VisitorKeys): VisitorKeys;

type VisitorKeys = VisitorKeys$1;

export { KEYS, VisitorKeys, getKeys, unionWith };

```

</details>

exported `KEYS` will be of type `VisitorKeys$1` instead of `VisitorKeys`. Is that okay?

**Comment by bradzacher** [comment]
> exported KEYS will be of type VisitorKeys$1 instead of VisitorKeys. Is that okay

Yes - the names are not required to be the same - the name is just an alias.
The types are resolved to their actual representation before they are checked by TS.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

Leaving open for others to review.

**Review by snitin315 — APPROVED** [review]
LGTM. Thanks for contributing :rocket:

**Pull request merged**